pkg/reconciler/pipelinerun/resources/pipelinerunresolution_test.go

```diff
@@ -972,32 +1007,38 @@ func TestIsSkipped(t *testing.T) {
 		},
 	}, {
 		name:  "run-started",
-		state: oneRunStartedState,
+		tasks: oneRunStartedState,
 		expected: map[string]bool{
 			"mytask13": false,
 		},
 	}, {
 		name:  "run-cancelled",
-		state: runCancelled,
+		tasks: runCancelled,
 		expected: map[string]bool{
 			"mytask13": false,
 		},
 	}} {
 		t.Run(tc.name, func(t *testing.T) {
-			d, err := dagFromState(tc.state)
+			d, err := dagFromState(tc.tasks)
+			if err != nil {
+				t.Fatalf("Could not get a dag from the TC state %#v: %v", tc.tasks, err)
+			}
+			f, err := dagFromState(tc.finally)
```

Comment:
there's something very odd about getting a "dag" from the finally tasks - but i think thats a whole other story XD

Reply:
created a PR for the refactor so that we consider them separately: https://github.com/tektoncd/pipeline/pull/3744